pkg/search/proxy/controller.go

```diff
@@ -99,8 +89,8 @@ func NewController(restConfig *restclient.Config, restMapper meta.RESTMapper,
 		},
 	}
 
-	karmadaFactory.Cluster().V1alpha1().Clusters().Informer().AddEventHandler(resourceEventHandler)
-	karmadaFactory.Search().V1alpha1().ResourceRegistries().Informer().AddEventHandler(resourceEventHandler)
+	option.KarmadaFactory.Cluster().V1alpha1().Clusters().Informer().AddEventHandler(resourceEventHandler)
+	option.KarmadaFactory.Search().V1alpha1().ResourceRegistries().Informer().AddEventHandler(resourceEventHandler)
 
 	return ctl, nil
```

Comment:
There is no error return, so maybe delete this parameter？

Reply:
OK